### PR TITLE
Support rook-pivoting in bkfact

### PR DESCRIPTION
**Pull request by timholy**
LAPACK 3.5 supports alternative [rook-pivoting algorithms](http://www.netlib.org/lapack/improvement.html) for Bunch-Kaufman. This PR wraps this functionality.

Best reviewed with `?w=1` appended to the URL.

### CODE REVIEW

File: base/linalg/lapack.jl [hunk 1]
```diff
@@ -3847,28 +3869,36 @@ for (syconv, hesv, hetrf, hetri, hetrs, elty, relty) in
...
         end
     end
 end
 
-for (sysv, sytrf, sytri, sytrs, elty, relty) in
-    ((:zsysv_,:zsytrf_,:zsytri_,:zsytrs_,:Complex128, :Float64),
-     (:csysv_,:csytrf_,:csytri_,:csytrs_,:Complex64, :Float32))
-    @eval begin
+for suffix in (symbol(""), :_rook)
+    for (sysv, sytrf, sytri, sytrs, elty, relty) in
```
Comment (andreasnoack) [inline base/linalg/lapack.jl]: I think it would be simpler to have a loop for each function given the convoluted code just below.

**Comment by andreasnoack** [comment]
Looks good to me except for the small style comment, maybe we should wait until https://github.com/JuliaLang/julia/pull/14309 has been merged.

**Comment by timholy** [comment]
Rebased and updated, including the style change. Curiously, Travis never ran this as a PR (it did run the push), but AppVeyor did, so hopefully we're OK.

**Comment by nalimilan** [comment]
It would be nice to make a bit more advertisement in the future when raising dependency requirements like that. It took me some time to realize why my EPEL RPM builds where failing.

**Comment by timholy** [comment]
Since I always build openblas from source, I didn't think about this as a potential problem. Sorry about that, and thanks for fixing!

**Comment by nalimilan** [comment]
You're welcome @timholy.

**Comment by ViralBShah** [comment]
Could this have been wrapped in a test for `laver()`? That would allow us to avoid making the minimum LAPACK at 3.5. The main reason is that users of MKL may not have new versions of LAPACK.

**Comment by ViralBShah** [comment]
Seems like `LAPACK.VERSION[]` is the new way to do this.

**Comment by ViralBShah** [comment]
I seem to recollect that even `dgesdd` was introduced in 3.5 or perhaps 3.4, and used to be protected with a version check once upon a time.

BTW, I am ok with 3.5 being the minimum, but just wanted to point out the issues.

**Comment by eschnett** [comment]
One can probably work around missing LAPACK functions in MKL with some linker magic. One needs to link against MKL explicitly (i.e. using an `-l` flag, not the `-mkl` flag), then link against a new LAPACK. This will preferably resolve symbols against MKL, and then take any missing symbols from LAPACK.

**Pull request merged**